include/pfs/parsers.hpp

```diff
@@ -80,13 +81,19 @@ netlink_socket parse_netlink_socket_line(const std::string& line);
 dev_t parse_device(const std::string& device_str);
 task_state parse_task_state(char state_char);
 
+// When parsing key value text files, we might encounter different keys which
+// should be parsed in the same manner. To relate those lines in the same
+// manner, a remap function can be used.
+using RemapFunction = const std::function<std::string(const std::string &)>;
+
 template <typename Output>
 class file_parser
 {
 public:
     Output parse(const std::string& path,
                  const char delim,
-                 const std::set<std::string>& keys = {})
+                 const std::set<std::string>& keys = {},
+                 RemapFunction& remap = nullptr)
```

Comment:
I called it key_remap, I think it's redundant to write the word function in the name, because the type already informs you about that, anyway your call.

Reply:
`key_remap` is great!